packages/uploadthing/src/internal/next/approuter.ts

```diff
@@ -20,16 +24,27 @@ export const createNextRouteHandler = <TRouter extends FileRouter>(
       actionType,
       req,
     });
-    if (response.status === 200) {
-      return new Response(JSON.stringify(response.body), {
-        status: response.status,
+
+    if (response instanceof UploadThingError) {
+      const formattedError = errorFormatter(response);
+      return new Response(JSON.stringify(formattedError), {
+        status: getStatusCodeFromError(response),
+        headers: {
+          "x-uploadthing-version": UPLOADTHING_VERSION,
+        },
+      });
+    }
+    if (response.status !== 200) {
```

Comment:
`if (response.status !== 200 || !response.message)` ? 

Since this logic is actually different from what it was previously. Or we can consider the way that it was checking before incorrect, and this is better.